docs/compendium/substitute.md

```diff
@@ -103,7 +105,8 @@ additional metadata of a substituted ERC
           "substitutionFiles": [
             {
               "original": "climate-timeseries.csv",
-              "xchange": "mytimeseries_data.csv"
+              "xchange": "mytimeseries_data.csv",
+              "filename": "climate-timeseries.csv"
```

Comment:
If original != xchange, then we do not need filename and it should not be there.

Normal case:

```
{
   "base":"climate-timeseries.csv",
   "overlay":"mytimeseries_data.csv"
}
```

And _if_ the base contains files `data.csv` and `input.csv`, and we want to overlay `data.csv` in the base with a file `input.csv` from another ERC (but we cannot simply replace base's `input.csv` because we do not know how it is still needed) then we need to rename `input.csv` and extend the overlay with a specific file to use:

```json
{
   "base":"data.csv",
   "overlay":"input.csv",
   "filename":"overlay_input.csv"
}
```

This allows us to handle three important things:

- the substituter can mount `../overlay_input.csv:/erc/data.csv`, i.e. replacing the data file in the container
- the UI can visually link base/data.csv with overlay/input.csv
- input.csv from base, which is not involved in the overlay, remains unchanged

Reply:
"additional metadata of a substituted ERC" is not a full sentence yet, please expand.

Also, please explain what `filename` means, and give more than one example (i.e. one without `filename`, and one with `filename`).